src/qt/rpcconsole.cpp

```diff
@@ -1163,8 +1163,12 @@ void RPCConsole::banSelectedNode(int bantime)
         // Find possible nodes, ban it and clear the selected node
         const CNodeCombinedStats *stats = clientModel->getPeerTableModel()->getNodeStats(detailNodeRow);
         if(stats) {
-            g_connman->Ban(stats->nodeStats.addr, BanReasonManuallyAdded, bantime);
-            g_connman->DisconnectNode(id);
+            if (g_banman) {
+                g_banman->Ban(stats->nodeStats.addr, BanReasonManuallyAdded, bantime);
+            }
+            if (g_connman) {
+                g_connman->DisconnectNode(id);
```

Comment:
Same response :)

Note that these comments are a bit stale. We used to track down the entry by looking through all node addresses, but now we use the id directly. I'll update those comments independently.

Also, see the behavior of disconnectSelectedNode above. I think it makes sense to be consistent with that?

For the sake of not over-complicating this, It may make sense to just keep the old behavior for this PR and switch to id in a follow-up.